src/stm32/fdcan.c

```diff
@@ -128,8 +145,14 @@ canbus_set_filter(uint32_t id)
     can_filter(0, CANBUS_ID_ADMIN);
     can_filter(1, id);
     can_filter(2, id + 1);
+
+#if CONFIG_MACH_STM32G0
     SOC_CAN->RXGFC = ((id ? 3 : 1) << FDCAN_RXGFC_LSS_Pos
                       | 0x02 << FDCAN_RXGFC_ANFS_Pos);
+#elif CONFIG_MACH_STM32H7
+    SOC_CAN->SIDFC |= 3 << FDCAN_SIDFC_LSS_Pos;
+    SOC_CAN->GFC = 0x02 << FDCAN_GFC_ANFS_Pos;
+#endif
```

Comment:
Out of curiosity, why does the stm32g0 code use `id ? 3 : 1` while the h7 unconditionally uses `3`?

Reply:
stm32h7 is also changed to `id ? 3 : 1`.